traffic_ops/traffic_ops_golang/deliveryservice/autorenewcerts.go

```diff
@@ -102,7 +102,8 @@ func renewCertificates(w http.ResponseWriter, r *http.Request, deprecated bool)
 		existingCerts = append(existingCerts, ExistingCerts{Version: ds.Version, XmlId: ds.XmlId})
 	}
 
-	ctx, _ := context.WithTimeout(r.Context(), AcmeTimeout*time.Duration(len(existingCerts)))
+	ctx, cancelTx := context.WithTimeout(r.Context(), AcmeTimeout*time.Duration(len(existingCerts)))
+	defer cancelTx()
```

Comment:
I'm not sure this is what we want, since this will cancel the context before the spawned goroutine on L113 (which uses the context) has a chance to complete its task.

Reply:
Cancelling the context in the goroutine in e55a35c622.